pkg/config/config.go

```diff
@@ -0,0 +1,149 @@
+// Copyright © 2018 The Helm Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"github.com/mitchellh/go-homedir"
+	"path"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/helm/chart-testing/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var (
+	homeDir, _ = homedir.Dir()
+	configSearchLocations = []string{"/etc/ct", path.Join(homeDir, "ct"), "."}
+)
+
+type Configuration struct {
+	Remote                string        `mapstructure:"remote"`
+	TargetBranch          string        `mapstructure:"target-branch"`
+	BuildId               string        `mapstructure:"build-id"`
+	TillerNamespace       string        `mapstructure:"tiller-namespace"`
+	LintConf              string        `mapstructure:"lint-conf"`
+	ChartYamlSchema       string        `mapstructure:"chart-yaml-schema"`
+	ValidateMaintainers   bool          `mapstructure:"validate-maintainers"`
+	CheckVersionIncrement bool          `mapstructure:"check-version-increment"`
+	ProcessAllCharts      bool          `mapstructure:"all"`
+	Charts                []string      `mapstructure:"charts"`
+	ChartRepos            []string      `mapstructure:"chart-repos"`
```

Comment:
I see for the format for ChartRepos is "incubator=https://incubator". Would it be more intuitive if those were broken up? Then the YAML would look something like:

```yaml
chart-repos:
  - name: incubator
    location: https://incubator
```

Reply:
All Cobra CLI flags are bound to Viper. This is generic. I guess changing this is not easily possible. And it would not be consistent.